src/main/java/org/elasticsearch/index/search/child/TopChildrenQuery.java

```diff
@@ -176,11 +179,16 @@ int resolveParentDocuments(TopDocs topDocs, SearchContext context, ObjectObjectO
                 }
                 if (parentDocId != DocIdSetIterator.NO_MORE_DOCS) {
                     // we found a match, add it and break
-                    IntObjectOpenHashMap<ParentDoc> readerParentDocs = parentDocsPerReader.get(indexReader.getCoreCacheKey());
+                    IntObjectHashMap<ParentDoc> readerParentDocs = parentDocsPerReader.get(indexReader.getCoreCacheKey());
                     if (readerParentDocs == null) {
-                        //The number of docs in the reader and in the query both upper bound the size of parentDocsPerReader
-                        int mapSize =  Math.min(indexReader.maxDoc(), context.from() + context.size());
-                        readerParentDocs = new IntObjectOpenHashMap<>(mapSize);
+                        // TODO: this is the previous comment and code:
```

Comment:
This snippet requires some knowledge of the internals -- the way it was previously is incorrect (negative estimated number of elements). I think the context shouldn't return invalid (-1) values the way it currently does.